conan/internal/graph/installer.py

```diff
@@ -253,12 +253,16 @@ def install(self, deps_graph, remotes, install_order=None):
         handled_count = 1
 
         self._download_bulk(install_order)
+        prefs_lru = []
         for level in install_order:
             for install_reference in level:
                 for package in install_reference.packages.values():
                     self._install_source(package.nodes[0], remotes)
                     self._handle_package(package, install_reference, handled_count, package_count)
                     handled_count += 1
+                    prefs_lru.append(package.nodes[0].pref)
```

Comment:
To maintain the previous behaviour, shouldn't we check at this point that `package.binary == BINARY_CACHE` before appending?